src/internal/helpers.jl

```diff
@@ -30,17 +30,11 @@ function evaluate_f!!(f::NonlinearFunction{iip}, fu, u, p) where {iip}
 end
 
 # AutoDiff Selection Functions
-function get_concrete_forward_ad(
-        autodiff::Union{ADTypes.AbstractForwardMode, ADTypes.AbstractFiniteDifferencesMode},
-        prob, sp::Val{test_sparse} = True, args...; kwargs...) where {test_sparse}
-    return autodiff
-end
 function get_concrete_forward_ad(
         autodiff::ADTypes.AbstractADType, prob, sp::Val{test_sparse} = True,
-        args...; check_reverse_mode = true, kwargs...) where {test_sparse}
```

Comment:
This change will need to be propaged to the calls to `get_concrete_forward_ad`

Reply:
Done. I also spotted some instances of the opposite error, where `check_forward_mode` was supplied to `get_concrete_reverse_ad`. I guess it must have been ignored instead of erroring because some of the functions there accept `kwargs...`